lib/state.js

```diff
@@ -510,53 +528,68 @@ StateResolver.prototype.storeObjectToVariableTable_ = function(obj) {
  *
  * See https://github.com/iojs/io.js/issues/1190.
  */
-StateResolver.prototype.resolveMirror_ = function(mirror) {
+StateResolver.prototype.resolveMirror_ = function(mirror, isEvaluated) {
   if (semver.satisfies(process.version, '<1.6')) {
-    return this.resolveMirrorSlow_(mirror);
+    return this.resolveMirrorSlow_(mirror, isEvaluated);
   } else {
-    return this.resolveMirrorFast_(mirror);
+    return this.resolveMirrorFast_(mirror, isEvaluated);
   }
 };
 
 // A slower implementation of resolveMirror_ which is safe for all node versions
-StateResolver.prototype.resolveMirrorSlow_ = function(mirror) {
+StateResolver.prototype.resolveMirrorSlow_ = function(mirror, isEvaluated) {
   // Instead, let's use Object.keys. This will only get the enumerable
   // properties. The other alternative would be Object.getOwnPropertyNames, but
   // I'm going with the former as that's what util.inspect does.
   var that = this;
+
   var keys = Object.keys(mirror.value());
-  if (that.config_.capture.maxProperties) {
-    keys = keys.slice(0, that.config_.capture.maxProperties);
+  var maxProps = that.config_.capture.maxProperties;
+
+  if (!isEvaluated && maxProps) {
+    keys = keys.slice(0, maxProps);
   }
   var members = keys.map(function(prop) {
-    return that.resolveMirrorProperty_(mirror.property(prop));
+    return that.resolveMirrorProperty_(mirror.property(prop), isEvaluated);
   });
 
+  var mirrorVal = mirror.value();
+  var len = mirrorVal && mirrorVal.length;
   return {
-    value: mirror.toText(),
+    value: mirror.toText() + 
+      ((typeof len === 'undefined') ? '' : ' of length ' + len),
     members: members
   };
 };
 
 // A faster implementation of resolveMirror_ which segfaults in node <1.6
 //
 // See https://github.com/iojs/io.js/issues/1190.
-StateResolver.prototype.resolveMirrorFast_ = function(mirror) {
-  var members = this.getMirrorProperties_(mirror).map(
-      this.resolveMirrorProperty_.bind(this));
+StateResolver.prototype.resolveMirrorFast_ = function(mirror, isEvaluated) {
+  var resMirrorProp = this.resolveMirrorProperty_.bind(this);
```

Comment:
Should we be using `.bind` here or a closure due to performance concerns across varying Node versions?
cc @ofrobots